txtorcon/socks.py

```diff
@@ -203,9 +200,9 @@ def _parse_request_reply(self):
             return
 
         if reply != self.SUCCEEDED:
-            try:
-                self.reply_error(_socks_reply_code_to_string[reply])
-            except KeyError:
+            if reply in SocksErrorFactory.socks_errors:
+                self.reply_error(reply)
+            else:
                 self.reply_error("Unknown SOCKS reply code {}".format(reply))
```

Comment:
I hated to put that check in there, but as `_disconnect` could receive either an error or a message that was the solution I found :/ Now that it is receiving an error, this looks a lot better while just catching the `KeyError`.